### ISSUE TITLE
next 14.2.4 type:module give error 

### ISSUE THREAD

**Issue by tianyingchun** [timlrx/contentlayer2#20]
Generated 13 documents in .contentlayer
 ✓ Ready in 3.7s
 ○ Compiling / ...
 ✓ Compiled / in 4.9s (1255 modules)
 GET / 200 in 5301ms
 ✓ Compiled in 340ms (580 modules)
<w> [webpack.cache.PackFileCacheStrategy/webpack.FileSystemInfo] Parsing of /Users/tianyingchun/Documents/hyperse-internal/hyperse-blog/node_modules/@contentlayer2/core/dist/generation/generate-dotpkg.js for build dependencies failed at 'import(URL.pathToFileURL(filePathJoin(generatedPkgPath, 'generated', 'index.mjs')).href)'.
<w> Build dependencies behind this expression are ignored and might cause incorrect cache invalidation.

**Comment by tianyingchun** [comment]
"contentlayer2": "0.4.6",
     "next-contentlayer2": "0.4.6",

**Comment by timlrx** [comment]
Sorry, could you elaborate on what causes the error and provide a small reproducible is possible?

**Comment by tianyingchun** [comment]
okay, i will find time to make a small reproduce repo

**Comment by yangchristina** [comment]
Am also getting this

**Comment by timlrx** [comment]
Could you include more information on the version of Next and React that you are using as well. I just tested on Next 15 canary and React 19 without issue - https://stackblitz.com/edit/github-ekmxur-ba82kk?file=package.json

**Comment by tianyingchun** [comment]
see here 
https://github.com/tianyingchun/tailwind-nextjs-starter-blog-issue
1. yarn install
2. yarn dev
3. see below
<img width="1011" alt="image" src="https://github.com/timlrx/contentlayer2/assets/2833251/7b9e944d-429b-4b92-9802-f0e9a00011ef">

**Comment by tianyingchun** [comment]
also see latest commit 
https://github.com/tianyingchun/tailwind-nextjs-starter-blog-issue/commit/4e8b693192f19c539ca465073052e15e7c426b08

**Comment by timlrx** [comment]
Ah this is the same issue as https://github.com/contentlayerdev/contentlayer/issues/313. It's a Webpack warning that can be ignored: https://github.com/webpack/webpack/pull/15688. Or if you don't use `"type": "module"`, you can avoid this error as well.

**Comment by tianyingchun** [comment]
yes, i know, but don't know why, i just switch to `turbo` & webpack in memory cache.

**Comment by ham-evans** [comment]
I'm getting this too with `"contentlayer2": "^0.5.0"`.  Definitely triggered by `"type": "module"` in package.json but would be awesome if there was a way to remove it.
